src/electron/adapters/electron-storage-adapter.ts

```diff
@@ -5,45 +5,27 @@ import { pick } from 'lodash';
 import { IndexedDBDataKeys } from '../../background/IndexedDBDataKeys';
 import { StorageAdapter } from '../../common/browser-adapters/storage-adapter';
 import { IndexedDBAPI } from '../../common/indexedDB/indexedDB';
-import { createDefaultLogger } from '../../common/logging/default-logger';
-import { Logger } from '../../common/logging/logger';
 
 export class ElectronStorageAdapter implements StorageAdapter {
-    constructor(private readonly indexedDBInstance: IndexedDBAPI, private logger: Logger = createDefaultLogger()) {}
+    constructor(private readonly indexedDBInstance: IndexedDBAPI) {}
 
-    public setUserData(items: Object, callback?: () => void): void {
-        this.indexedDBInstance
-            .setItem(IndexedDBDataKeys.installation, items)
-            .then(() => callback && callback())
-            .catch(error => {
-                this.logger.error('Error occurred when trying to set user data: ', error);
-            });
+    public async setUserData(items: Object): Promise<void> {
+        await this.indexedDBInstance.setItem(IndexedDBDataKeys.installation, items);
     }
 
-    public getUserData(keys: string[], callback: (items: { [key: string]: any }) => void): void {
-        this.indexedDBInstance
-            .getItem(IndexedDBDataKeys.installation)
-            .then(data => {
-                const filteredData = pick(data, keys);
-                callback(filteredData);
-            })
-            .catch(error => {
-                this.logger.error('Error occurred when trying to get user data: ', error);
-            });
+    public getUserData(keys: string[]): Promise<{ [key: string]: any }> {
+        return this.indexedDBInstance.getItem(IndexedDBDataKeys.installation).then(data => pick(data, keys));
     }
 
-    public removeUserData(key: string): void {
-        this.indexedDBInstance
-            .getItem(IndexedDBDataKeys.installation)
-            .then(data => {
-                const filtered = Object.keys(data)
-                    .filter(internalKey => internalKey !== key)
-                    .reduce((obj, k) => {
-                        obj[key] = data[k];
-                        return obj;
-                    }, {});
-                this.indexedDBInstance.setItem(IndexedDBDataKeys.installation, filtered).catch(e => this.logger.error(e));
-            })
-            .catch(error => this.logger.error('Error occurred when trying to remove user data: ', error));
+    public async removeUserData(key: string): Promise<void> {
+        await this.indexedDBInstance.getItem(IndexedDBDataKeys.installation).then(data => {
```

Comment:
I think it's a little unexpected to see `await` and `.then(...)` syntaxes being mixed together in one function; maybe use await for both cases?

Reply:
I was debating about that... you just tilt the balance into async/await syntax. Updated